sqlx-core/src/arguments.rs

```diff
@@ -16,6 +16,10 @@ pub trait Arguments<'q>: Send + Sized + Default {
     fn add<T>(&mut self, value: T)
     where
         T: 'q + Send + Encode<'q, Self::Database> + Type<Self::Database>;
+
+    fn place_holder(&self, _argument_count: u16) -> String {
```

Comment:
Nit: `argument_count` shouldn't be necessary as the `Arguments` impl should already know how many arguments have been pushed. Personally, I also think "placeholder" should be one word, not two.

The intermediate `String` is also a wasted allocation. An alternative signature might look like this:
```rust
use std::fmt::{Self, Write};

fn format_placeholder<W: Write>(&self, mut writer: W) -> fmt::Result {
    writer.write_str("?")
}
```

and the Postgres implementation would look like this:

```rust
fn format_placeholder<W: Write>(&self, mut writer: W) -> fmt::Result {
    write!(writer, "${}", self.count)
}
```

`QueryBuilder` would then simply invoke it like:
```rust
// This is highly unlikely to return an error so panicking is fine.
arguments.format_placeholder(&mut self.query).expect("error in format_placeholder")
```